lib/streamlit/DeltaGenerator.py

```diff
@@ -2141,6 +2167,20 @@ def file_uploader(
         if isinstance(type, str):
             type = [type]
 
+        encoding = kwargs.get("encoding")
+        has_encoding = "encoding" in kwargs
+        show_deprecation_warning = config.get_option(
+            "deprecation.v00_063_showfileUploaderEncoding"
+        )
+
+        if show_deprecation_warning and (
+            (has_encoding and encoding is not None) or not has_encoding
```

Comment:
I might be misunderstanding this, but why do we show the deprecation warning when the user did not provide an encoding?

Reply:
If no encoding is provided, we used to default to an auto encoding for them. We wanted to show them the warning in case they were relying on the auto encoding and expecting a StringIO through the auto encoding. This could be showing up unnecessarily for folks that are not dealing with StringIOs